src/store.h

```diff
@@ -110,6 +114,10 @@ class MetadataStore {
   mutable std::mutex metadata_mu_;
   // A map from MonitoredResource to (JSON) resource metadata.
   std::map<MonitoredResource, Metadata> metadata_map_;
+  // A lock that guards access to the last collection map.
+  mutable std::mutex last_collection_mu_;
+  // A map from resource type to last collection time.
+  std::map<std::string, Timestamp> last_collection_map_;
```

Comment:
Technically, this is just last collection times by object type, so I would suggest calling this variable `last_collection_times_` (and modify the accessor accordingly to `GetLastCollectionTimes`).